rg_instructor_analytics/views/TabFragment.py

```diff
@@ -33,13 +35,52 @@ def process(self, request, **kwargs):
         if enroll_end is None:
             enroll_end = course.end
 
-        enroll_info = {
+        return {
             'enroll_start': mktime(enroll_start.timetuple()) if enroll_start else 'null',
             'enroll_end': mktime(enroll_end.timetuple()) if enroll_end else 'null',
         }
+
+    def get_avalibel_courses(self, user):
+        """
+        Return courses, available for the given user.
+        """
+        avalibel_courses = CourseAccessRole.objects.filter(user=user, role__in=['instructor', 'staff'])
+        exist_courses_id = []
+        result = []
+        for record in avalibel_courses:
+            try:
+                course = get_course_by_id(record.course_id, depth=0)
+                course_id = str(course.id)
```

Comment:
why do we need course_id if we can check whether current `course` is already added to `result` dict?

Reply:
Because 2 CourseKey objects for the same course has a different hash.

Reply:
Got it, thanks.